src/core.rs

```diff
@@ -674,7 +683,7 @@ impl<T> Webauthn<T> {
     pub fn generate_challenge_authenticate_extensions(
         &self,
         creds: Vec<Credential>,
-        extensions: Option<JSONExtensions>
+        extensions: Option<JSONExtensions>,
```

Comment:
Should this also be of the new extensions input type? It wasn't super clear to me if there should be different extension types for different parts of the protocol.

Reply:
I think it should be different to indicate that the extensions may only be valid in registration or assertion contexts yes.

Reply:
We haven't implemented extensions for authentication, so are we good leaving this as `Option<JSONExtensions>`

Reply:
we should rename it to be something clearer like `Option<RequestAssertionExtension>`. At the same time we probably want to make a similar name change for `Option<RequestRegistrationExtension>`.

Reply:
What is `Option<RequestRegistrationExtension>`?

Reply:
Rename `extensions: Option<AuthenticatorExtensionsClientInputs>,` to `Option<RequestRegistrationExtension>` is what I mean.

Reply:
Okay. Do you think there might be a better word than `Request`? I worry a bit because although from the RP's (and thus `webauthn-rs`'s) perspective this is like a "request" we send to the client, it is technically a "response" in the world of http(s).

Reply:
Perhaps `RegistrationExtensionInputs`s? I'll also note in case it wasn't clear that I got the name `AuthenticatorExtensionsClientInputs` from the spec.

Reply:
Also not to get too fussy but could singularizing extensions to extension perhaps confuse users who think that the struct will be the input data for a single extension?

Reply:
I'd say that the terminology used in the Webauthn spec is "a hot mess" at best, and generally incomprehensible to most people (sometimes even leading to security issues but you know ... ). So sometimes it can be best to avoid it, and to use human accessible names, and just put in our code documentation that "this is the X term from the webauthn spec".

Anyway, the reason I said Option<RequestRegistrationExtensions> is because "we are requesting these extensions to be added by the client when it signs this registration". So I think from that direction the term Request here makes sense. 

Alternately, just "RegistrationExtensions" would work too.

Also yes, it should be plural, I think I mistyped.

Reply:
Ah, I see. Then what would you suggest naming the outputs struct? `RegistrationExtensionsOutput`?

Reply:
SignedRegistrationExtensions?
RegistrationExtensionsResponse?
FlargbolAbloby?

Reply:
I like that last one XD

Reply:
Yolo it to production then :D 

More seriously, I think SignedRegistrationExtensions is the best here, because it indicates clearly that it's different, and that they can be trusted by the RP.

Reply:
Sounds good—I guess we'd just need to be careful that we never spit that struct back out at a user until we've actually validated the signature

Reply:
We would only return this as part of attested credential data after a full and complete verification through our code anyway, so I think that this is fine.

Reply:
Also, since this is still 0.3 alpha, worst case we can rename it anyway because we can still make breaking changes. 

I'd actually like to merge this today as I want to work on some other changes of the api today/tomorrow, so did you want to do the tests in a second pr?

Reply:
Sounds good. I'll push the renaming changes and PR the tests later